src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorBuildStep.groovy

```diff
@@ -104,12 +130,26 @@ class IqPolicyEvaluatorBuildStep
     }
 
     @Override
-    FormValidation doCheckIqApplication(@QueryParameter String value) {
+    FormValidation doCheckListAppId(@QueryParameter String value) {
       FormValidation.validateRequired(value)
     }
 
     @Override
-    ListBoxModel doFillIqApplicationItems(@QueryParameter String jobCredentialsId, @AncestorInPath Job job) {
+    FormValidation doCheckManualAppId(@QueryParameter String value, @QueryParameter String jobCredentialsId,
+                                      @AncestorInPath Job job)
+    {
+      FormValidation val = FormValidation.validateRequired(value)
+      if (FormValidation.ok() == val) {
+        if (!IqUtil.
```

Comment:
This will create a new application every time the value is checked by the UI, I don't think we want that.

Reply:
> This will create a new application every time the value is checked by the UI, I don't think we want that.

I see what you mean.  Not a good place.  Need to move it.  How do I show an error and stop the submit?  Is throwing an exception the only way?  The GUI form validation was cool because it created an error widget.  Throwing an exception shows a stack trace.

Reply:
Since we only know the error once the build kicks off, exception + stack trace is the only way.